src/router/BodLayerId.routerPlugin.js

```diff
@@ -0,0 +1,236 @@
+        }
+        store.dispatch('setCenter', {
+            center: center,
+            source: 'Bod Layer Id router',
+        })
+    }
+    if (!bodIdParams.showToolTip) {
+        // we'll need to see how to do this better, as this doesn't  work
+        store.dispatch('hideToolTip') // this is apparently not the right thing to show
+    }
+}
+
+/**
+ * @param {[Number[], Number[]]} extent An array of two coordinates
+ * @param {Number[]} center The coordinates to the center of the extent. We could calculate it
+ *   inside, but it's already done
+ * @param {Store} store
+ * @returns {Number} The zoom level at which we can see the whole extent
+ */
+export function getZoomFromExtent(extent, center, store) {
+    const extentSize = {
+        width: Math.abs(extent[1][0] - extent[0][0]),
+        height: Math.abs(extent[1][1] - extent[0][1]),
+    }
+    let targetResolution
+    // if the extent's height is greater than width, we base our resolution calculation on that
+    if (extentSize.height > extentSize.width) {
+        targetResolution = extentSize.height / (store.state.ui.height - 96) /* header height */
+    } else {
+        targetResolution = extentSize.width / store.state.ui.width
+    }
+
+    const zoomForResolution = store.state.position.projection.getZoomForResolutionAndCenter(
+        targetResolution,
+        center
+    )
+
+    return Math.max(zoomForResolution - 1, 0)
+}
+
+/**
+ * Parse the URL for bod-layer-id parameters (with or without the showToolTip parameter), remove the
+ * parameter for the URL while dispatching what is needed to the store
+ *
+ * Example:
+ *
+ * - `http://localhost:8080/#/?geolocation=true&some.layer_id=213,234` =>
+ *   `http://localhost:8080/#/?geolocation=true&layers=some.layer_id`
+ *
+ * @param {Router} router
+ * @param {Store} store
+ */
+
+const bodLayerIdRouterPlugin = (router, store) => {
+    let changesMadeByThisPlugin = false
+    router.beforeEach((to, from, next) => {
+        /**
+         * We need to ensure the startup is finished to handle the bod layer Id, as we need some
+         * data to be defined (most importantly : the layers config)
+         */
+
+        if (to.name === 'MapView') {
+            if (!changesMadeByThisPlugin) {
+                // maybe use to.query instead of window.location.hash
+                const bodIdParams = window.location?.hash
+                    ? parseBodLayerParamsFromQuery(to.query, store.state.layers.config)
+                    : null
+                if (bodIdParams?.layer) {
+                    changesMadeByThisPlugin = true
+                    handleBodIdParams(bodIdParams, store, to, next)
+                } else {
+                    next()
+                }
+            } else {
+                changesMadeByThisPlugin = false
+                next()
+            }
+        } else {
+            changesMadeByThisPlugin = false
+            next()
+        }
+    })
```

Comment:
This should be simplified, not using the `next()`.
I would also remove the `changesMadeByThisPlugin` flag, this would mean that parameter lookup would be made twice, but it will be anyway done for every route changes. It make also sense to register this plugin after the router store sync, at the end.